lib/webmock/http_lib_adapters/httpclient_adapter.rb

```diff
@@ -186,10 +186,15 @@ def build_request_signature(req, reuse_existing = false)
     end
     headers = headers_from_session(uri).merge(headers)
 
-    if (auth_cred = auth.get(req)) && auth.scheme == 'Basic'
+    if ((auth_cred = auth.get(req)) && auth.scheme == 'Basic') || auth_cred = headers.select {|k,v| k =~ /[Aa]uthorization/ && v.select { |v2| v2 =~ /^Basic / }.first }.values.first.first
```

Comment:
Possibly the `.values.first.first` can be solved in a nicer way, but i was a bit lazy. It might also break when `headers = {"Authorization": []}`